index.js

```diff
@@ -21,18 +38,37 @@ async function main () {
   password = password === '' ? undefined : password
   console.log(`connecting to ${host} ${port} with ${username}`)
 
+  status = 'Logging in'
+
+  document.getElementById('loading-text').requestFullscreen()
+
   const bot = mineflayer.createBot({
     host,
     port,
     username,
     password
   })
 
+  bot.once('error', () => {
+    console.log('Encountered error!')
+    status = 'Error encountered. Please reload the page'
+  })
+
+  bot.once('kicked', () => {
+    console.log('User was kicked!')
+    status = 'The Minecraft server kicked you. Please reload the page to rejoin'
```

Comment:
this isnt sent because it encounters an error prior to this, just put the message user was disconnected on the bot.once('end') event

Reply:
Even after changing this, it still doesn't work. I'm giving up on this for now. It's going to be pretty rare of an issue, and spending any more time which could be used on adding crucial features such as inventory isn't really worth it any more.